pkg/entity/container_test.go

```diff
@@ -75,9 +94,199 @@ func TestContainer_GetMemory_Unsuccessful(t *testing.T) {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
 			_, err := tt.res.GetMemory()
 
-			if err == nil {
-				t.Error("return error of GetMemory does not match expected error")
-			}
+			assert.NotEqual(t, err, nil)
+		})
+	}
+}
+
+func TestContainer_NoLimits(t *testing.T) {
+	var tests = []struct {
+		res      Container
+		expected bool
+	}{
+		{
+			res: Container{
+				Memory: "",
+				Cpus:   "",
+			},
+			expected: true,
+		},
+		{
+			res: Container{
+				Memory: "5gb",
+				Cpus:   "",
+			},
+			expected: false,
+		},
+		{
+			res: Container{
+				Memory: "",
+				Cpus:   "5",
+			},
+			expected: false,
+		},
+		{
+			res: Container{
+				Memory: "4gb",
+				Cpus:   "6",
+			},
+			expected: false,
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			assert.Equal(t, tt.res.NoLimits(), tt.expected)
+		})
+	}
+}
+
+func TestContainer_NoCPULimits(t *testing.T) {
+	var tests = []struct {
+		res      Container
+		expected bool
+	}{
+		{
+			res: Container{
+				Cpus: "",
+			},
+			expected: true,
+		},
+		{
+			res: Container{
+				Cpus: "5",
+			},
+			expected: false,
+		},
+		{
+			res: Container{
+				Cpus: " ",
+			},
+			expected: false,
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			assert.Equal(t, tt.res.NoCPULimits(), tt.expected)
+		})
+	}
+}
+
+func TestContainer_NoMemoryLimits(t *testing.T) {
+	var tests = []struct {
+		res      Container
+		expected bool
+	}{
+		{
+			res: Container{
+				Memory: "",
+			},
+			expected: true,
+		},
+		{
+			res: Container{
+				Memory: " ",
+			},
+			expected: false,
+		},
+		{
+			res: Container{
+				Memory: "5GB",
+			},
+			expected: false,
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			assert.Equal(t, tt.res.NoMemoryLimits(), tt.expected)
+		})
+	}
+}
+
+func TestContainer_GetEnv(t *testing.T) {
+	var tests = []struct {
+		c        Container
+		expected []string
+	}{
+		{
+			c: Container{
+				Environment: map[string]string{"test": "env"},
+			},
+			expected: []string{"test=env"},
+		},
+		{
+			c: Container{
+				Environment: *new(map[string]string),
+			},
+			expected: *new([]string),
```

Comment:
[]string{}?

Reply:
So this will fail because `[]string{}` literal is not a default `nil` value, which is what's expected according to function logic